arrow/src/compute/kernels/comparison.rs

```diff
@@ -898,6 +899,224 @@ pub fn gt_eq_utf8_scalar<OffsetSize: StringOffsetSizeTrait>(
+            //     dyn_cmp_scalar!($LEFT, right, UInt64Array, $OP, UInt64Type)
+            // }
+            // (DataType::Float32, DataType::Float32) => {
+            //     dyn_cmp_scalar!($LEFT, right, Float32Array, $OP, Float32Type)
+            // }
+            // (DataType::Float64, DataType::Float64) => {
+            //     dyn_cmp_scalar!($LEFT, right, Float64Array, $OP, Float64Type)
+            // }
+            // (DataType::Utf8, DataType::Utf8) => {
+            //     let right: i32 = right.try_into().map_err(|_| {
+            //         ArrowError::ComputeError(String::from(
+            //             "Can not convert scalar to i128",
+            //         ))
+            //     })?;
+            //     let left =
+            //         $LEFT
+            //             .as_any()
+            //             .downcast_ref::<StringArray>()
+            //             .ok_or_else(|| {
+            //                 ArrowError::CastError(String::from(
+            //                     "Left array cannot be cast",
+            //                 ))
+            //             })?;
+            //     $OP::<Int32Type>(left, right)
+            // }
+            // (DataType::LargeUtf8, DataType::LargeUtf8) => {
+            //     dyn_cmp_scalar!($LEFT, $RIGHT, LargeStringArray, $OP, i64)
+            // }
+            (t1, t2) if t1 == t2 => Err(ArrowError::NotYetImplemented(format!(
+                "Comparing arrays of type {} is not yet implemented",
+                t1
+            ))),
+            (t1, t2) => Err(ArrowError::CastError(format!(
+                "Cannot compare an array with a scalar of different type ({} and {})",
+                t1, t2
+            ))),
+        }
+    }};
+}
+
+pub trait IntoArrowNumericType {
+    // type Arrow: ArrowNumericType<Native = Self>;
+    fn get_arrow_type(&self) -> &DataType;
+}
+
+impl IntoArrowNumericType for i8 {
+    // type Arrow = Int8Type;
+    fn get_arrow_type(&self) -> &DataType {
+        &DataType::Int8
+    }
+}
+
+impl IntoArrowNumericType for i16 {
+    // type Arrow = Int8Type;
+    fn get_arrow_type(&self) -> &DataType {
+        &DataType::Int16
+    }
+}
+
+impl IntoArrowNumericType for i32 {
+    // type Arrow = Int8Type;
+    fn get_arrow_type(&self) -> &DataType {
+        &DataType::Int32
+    }
+}
+
+/// Perform `left == right` operation on an array and a numeric scalar
+/// value. Supports PrimitiveArrays, and DictionaryArrays that have primitive values
+pub fn eq_dyn_scalar<T>(left: &dyn Array, right: T) -> Result<BooleanArray>
+where
+    T: IntoArrowNumericType + TryInto<i128> + Copy + std::fmt::Debug,
+{
+    let result = match left.data_type() {
+        DataType::Dictionary(key_type, _) => match key_type.as_ref() {
+            DataType::UInt8 => {
+                let left = left
+                    .as_any()
+                    .downcast_ref::<DictionaryArray<UInt8Type>>()
+                    .unwrap();
```

Comment:
You might be able to use https://docs.rs/arrow/6.4.0/arrow/array/fn.as_dictionary_array.html here too:

```suggestion
                let left = as_dictionary_array<DictionaryArray<UInt8Type>>()?;
```